D/DynarePreprocessor/build_tarballs.jl

```diff
@@ -0,0 +1,69 @@
+using BinaryBuilder
+
+env = copy(ENV)
+env["BINARYBUILDER_AUTOMATIC_APPLE"] = "true"
+env["BINARYBUILDER_USE_CCACHE"] = "true"
+
+name = "DynarePreprocessor"
+version = v"4.8.0-5"
+sha="a8fce06dc46ca09329e6899e1fde47f2cd81809b"
+sources = [
+    GitSource("https://git.dynare.org/Dynare/preprocessor.git", "a8fce06dc46ca09329e6899e1fde47f2cd81809b"),
+]
+
+script = raw"""
+if [[ ${target} == *-apple-* ]]
+then
+    sha="a8fce06dc46ca09329e6899e1fde47f2cd81809b"
+    gitlabjob="37258"
+    wget -O artifacts.zip https://git.dynare.org/Dynare/preprocessor/-/jobs/${gitlabjob}/artifacts/download
+    mkdir work
+    unzip artifacts.zip -d work
+    tar zxf work/${sha}/macos-x86_64/dynare-preprocessor.tar.gz
+    mkdir -p ../destdir/
+    mkdir -p ../destdir/bin
+    cp -r dynare-preprocessor ../destdir/bin
+    install_license ${WORKSPACE}/srcdir/preprocessor/COPYING
+else 
+    apk add samurai
+    apk add boost-dev
+    apk add bison=3.7.6-r0 --update-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/main 
+    apk add flex-dev
+
+    cd ${WORKSPACE}/srcdir/preprocessor
+
+    # remove -lstdc++fs in Makefile.am
+    sed s/-lstdc++fs// -i src/Makefile.am
+
+    autoreconf -si
+
+    update_configure_scripts
+    echo ${MACHTYPE}
+    ./configure --prefix=$prefix  --build=${MACHTYPE} --host=${target} --disable-doc LDFLAGS="-static -static-libgcc -static-libstdc++"
```

Comment:
Why the full static build?

Reply:
At least under Windows, I believe that end users won't have `mingw` installed and wouldn't have the dynamic libraries. Under Linux, it may avoid problems with older libraries on the end user system.

Reply:
Julia comes with the relevant compiler libraries :slightly_smiling_face:  For example Julia itself for Windows is built with the MinGW toolchain.

When you run the executable products of a JLL package using the [intended API](https://docs.binarybuilder.org/stable/jll/#ExecutableProduct), we make sure that those libraries are accessible by the program.  We go a long way to make sure static building isn't necessary :wink:

Reply:
Great. I will remove the static build